provider/resources.go

```diff
@@ -1246,6 +1246,45 @@ compatibility shim in favor of the new "name" field.`)
 						Name: "tags",
 					},
 				},
+				TransformOutputs: (func() func(ctx context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
+					r, ok := p.ResourcesMap().GetOk("aws_autoscaling_group")
+					contract.Assertf(ok, "Expected aws_autoscaling_resource to be defined")
+					defaultValue := func(rawPropName string) resource.PropertyValue {
+						prop, ok := r.Schema().GetOk(rawPropName)
+						contract.Assertf(ok, "Expected property %q to be defined", prop)
+						def := prop.Default()
+						contract.Assertf(def != nil, "Expected property %q Default() to be defined", prop)
+						switch prop.Type() {
+						case shim.TypeString:
+							defs, ok := def.(string)
+							contract.Assertf(ok, "Expected property %q Default() to be a string", prop)
+							return resource.NewStringProperty(defs)
+						case shim.TypeBool:
+							defb, ok := def.(bool)
+							contract.Assertf(ok, "Expected property %q Default() to be a bool", prop)
+							return resource.NewBoolProperty(defb)
+						default:
+							contract.Failf("Types other than string and bool are not supported yet")
+							return resource.NewNullProperty()
+						}
+					}
+					defaults := map[string]resource.PropertyValue{
+						"forceDelete":                    defaultValue("force_delete"),
```

Comment:
Can we detect the props that need the defaults to be fixed? This would help us generalize this and also make maintenance a bit easier (no need to play catchup with upstream).

Reply:
I think this works so well we should maybe change the bridge default behavior here in https://github.com/pulumi/pulumi-terraform-bridge/issues/2436 so that's how it works by default without the custom code.